Shopify/dawn
# [Image with Text] Add image masks

### PR Summary: 

Adds seven image masks to the Image with Text section (Circle, Arch, Parallellogram, Diamond, Right Chevron, Left Chevron, Blobs). 

### Why are these changes introduced?

These shapes introduce more design possibilities to our merchants. 

### What approach did you take?

These are implemented using CSS `clip-path()` or SVG mask shapes, as implemented in https://github.com/Shopify/dawn/pull/2302. 

### Other considerations

- The Round shape forces a circle. If the merchant adds a very tall image and has the "Adapt to Image" image height selected, this can look a little weird. It's easily fixable by selecting another image height option though. 
- When an image shape is active, the assigned background color for the section now extends below the image. We think this is more in line with what merchants want when they select a color background.
  - The global border and shadow controls for Media all behave as expected when this happens too.
  - Under these circumstances, a little bit of padding is added so the image shape doesn't bump up against the edges of the container. I've used the same padding that the text uses, so there's some symmetry there.
  - The color background, border, and shadow are not present if the "Overlap" content layout is active since they'd negatively interfere with this effect.
- The "Background 1" scheme is considered a special case. When this scheme is active, no color background is applied, and the border and shadow are suppressed. The reasoning here is that these generally tend to look bad/broken, and we think it's especially important that the default experience here works.
 
### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 | Do not include image hovers for the Image and Text block |  Implement animated hover states for the image when a shape is applied. | The images in that block are not clickable, and the hovers give visitors the impression that they are clickable. [We can re-assess this later](https://github.com/Shopify/dawn/pull/2302#discussion_r1107678028) and perhaps implement the animations on scroll instead of on hover. | It's just a little less fun. |

### Visual impact on existing themes

These shapes are off by default, and are optional. I don't expect any issues using them on other Dawn-based themes. 

### Testing steps/scenarios

1. [Visit the demo store.](https://os2-demo.myshopify.com/admin/themes/139389894678/editor)
2. Add a **Image with Text** section. 
3. In the section's settings panel, use the "Image Shape" dropdown to change the image shape.
4. Try all 7 of the supplied image shapes. 
5. Change other Section settings, to ensure they don't interfere with the image shape:
  - Image Height, Width
  - Image placement
  - Content layout (Overlap)
  - Color scheme
6. Scale the viewport from desktop down to mobile, ensuring that the shapes behave as intended with no clipping. 
7. Visit Theme Settings > Content Containers. Adjust Border & Shadow settings and ensure that these do not have unexpected results when combined with the image shape. 
8. Visit Theme Settings > Media. Adjust Border & Shadow settings and ensure that these do not have unexpected results when combined with the image shape. 

### Checklist
